### ISSUE TITLE
Next Up items do not always open the correct page

### ISSUE THREAD

**Issue by fice-t** [jellyfin/jellyfin-androidtv#3536]
### Describe the bug

This bug is similar to some other "Next Up" bugs (such as #3527), in that the root cause may be similar (or the same), but the result is different. I did in fact encounter #3527 once when working out a reproduction recipe for this bug, but I could not consistently trigger that bug.

The below, however, reproduces _this_ bug consistently on my device:

1. Play to completion an episode in a series, such that there is a valid Next Up item candidate.
2. Exit Jellyfin and re-enter Jellyfin, or otherwise ensure that this Next Up item appears in the home screen.
3. Play to completion an episode in a **_DIFFERENT_** series, such that there is a second valid Next Up item candidate.
4. Immediately upon finishing the episode, press the Home button to exit Jellyfin.
5. Immediately after exiting Jellyfin, select the Next Up item produced in Steps 1+2.

Expected: Jellyfin opens the page for the episode shown for the Next Up item from Steps 1+2.

Actual: Jellyfin opens the page for the next episode for the series from Step 3.

Perhaps Jellyfin is using the last page as the destination instead of the correct destination? Or perhaps it fails to switch to the new destination properly?

**Note**: The OP of  #3527 mentioned:
> Sometimes it helps to close the app and choose again the same Episode from the Next Up Screen

Similarly, if this (i.e. Step 2) is done again after Step 3, then this bug does **_NOT_** appear. That is, it seems to appear only if choosing a different Next Up item (from the Home screen) after a new item is added internally without it having been propagated to the Home screen.

### Application version

0.16.10

### Where did you install the app from?

Google Play

### Device information

Proprietary ATV Box

### Android version

Android 10

### Jellyfin server version

10.8.13

**Comment by fice-t** [comment]
Update with logs: I'm finding it a bit harder to reproduce in a emulator, but I managed to get logs of both expected Next Up behaviour and the buggy behaviour.


<details open><summary>Buggy behaviour</summary>
<p>

```log
02:51:23.839 ACRA                    org.jellyfin.androidtv.debug         D  onActivityPaused class org.jellyfin.androidtv.ui.browsing.MainActivity
02:51:24.533 ACRA                    org.jellyfin.androidtv.debug         D  onActivityStopped class org.jellyfin.androidtv.ui.browsing.MainActivity
02:51:24.535 MainActivity$onStop     org.jellyfin.androidtv.debug         D  MainActivity stopped
02:51:24.535 SessionRep...oreSession org.jellyfin.androidtv.debug         I  Restoring session
02:51:24.536 ACRA                    org.jellyfin.androidtv.debug         D  onActivitySaveInstanceState class org.jellyfin.androidtv.ui.browsing.MainActivity
02:51:30.703 ACRA                    org.jellyfin.androidtv.debug         D  onActivityStarted class org.jellyfin.androidtv.ui.browsing.MainActivity
02:51:30.707 ACRA                    org.jellyfin.androidtv.debug         D  onActivityResumed class org.jellyfin.androidtv.ui.browsing.MainActivity
02:51:30.988 jellyfin-apiclient      org.jellyfin.androidtv.debug         D  Adding request to queue: http://192.168.0.100:8096/Users/073b9d3f-c9ad-42bd-856d-e16644f2366d/Items/bb4f1ccb-e8eb-22e1-708c-aec9c8563f55?format=json
02:51:31.112 RecyclerView            org.jellyfin.androidtv.debug         W  No adapter attached; skipping layout
02:51:31.463 jellyfin-apiclient      org.jellyfin.androidtv.debug         I  Response received from: http://192.168.0.100:8096/Users/073b9d3f-c9ad-42bd-856d-e16644f2366d/Items/bb4f1ccb-e8eb-22e1-708c-aec9c8563f55?format=json
02:51:31.897 jellyfin-apiclient      org.jellyfin.androidtv.debug         D  Adding request to queue: http://192.168.0.100:8096/Shows/30fa6f894a5a8487d4342bd00ed59949/Episodes?AdjacentTo=bb4f1ccbe8eb22e1708caec9c8563f55&UserId=073b9d3f-c9ad-42bd-856d-e16644f2366d&format=json
02:51:31.918 FullDetailsFragment     org.jellyfin.androidtv.debug         D  Item type: Episode
02:51:31.920 jellyfin-apiclient      org.jellyfin.androidtv.debug         D  Adding request to queue: http://192.168.0.100:8096/Users/073b9d3f-c9ad-42bd-856d-e16644f2366d/Items?ParentId=9b95feb053de83f8bc7c01ff3658c859&StartIndex=2&recursive=false&Fields=PrimaryImageAspectRatio%2COverview%2CItemCounts%2CDisplayPreferencesId%2CChildCount&Limit=20&IncludeItemTypes=Episode&format=json
```


</p>
</details> 

<details open><summary>Expected (usual) behaviour</summary>
<p>

```log
02:55:23.565 ACRA                    org.jellyfin.androidtv.debug         D  onActivityPaused class org.jellyfin.androidtv.ui.browsing.MainActivity
02:55:24.188 ACRA                    org.jellyfin.androidtv.debug         D  onActivityStopped class org.jellyfin.androidtv.ui.browsing.MainActivity
02:55:24.190 MainActivity$onStop     org.jellyfin.androidtv.debug         D  MainActivity stopped
02:55:24.191 SessionRep...oreSession org.jellyfin.androidtv.debug         I  Restoring session
02:55:24.191 ACRA                    org.jellyfin.androidtv.debug         D  onActivitySaveInstanceState class org.jellyfin.androidtv.ui.browsing.MainActivity
02:55:27.602 ActivityTh...tensionsKt org.jellyfin.androidtv.debug         I  Applying theme DARK
02:55:27.602 ACRA                    org.jellyfin.androidtv.debug         D  onActivityCreated class org.jellyfin.androidtv.ui.startup.StartupActivity
02:55:27.630 ACRA                    org.jellyfin.androidtv.debug         D  onActivityStarted class org.jellyfin.androidtv.ui.startup.StartupActivity
02:55:27.634 ACRA                    org.jellyfin.androidtv.debug         D  onActivityResumed class org.jellyfin.androidtv.ui.startup.StartupActivity
```

</p>
</details>

The line of `onActivitySaveInstanceState` in each of the above is the last line before pressing the Home button, and all the following lines occur after selecting the Next Up item in the ATV home screen.

The logs indicate that in the buggy case, using the Next Up item to open Jellyfin doesn't start/create `StartupActivity` as usual, but instead immediately starts `MainActivity`. This means that the intent processing in `StartupActivity.openNextActivity` is skipped.

**Comment by nielsvanvelzen** [comment]
Although I haven't been able to reproduce this issue, I think I fixed it in #3612.

**Comment by fice-t** [comment]
I haven't been able to reproduce anymore in an emulator with Jellyfin on the tip of the `release-0.16.z`, though I've found it quite hard to reproduce in the emulator (without the patch) since my last post.

It's very nice to see that there will be another version released with 10.8 server support and that this will be in it.

**Comment by fice-t** [comment]
Update: after using `0.16.11` for several days, I haven't seen the issue reappear on my ATV device. Thanks again!
